Std/Data/List/Lemmas.lean

```diff
@@ -1156,6 +1162,17 @@ theorem drop_eq_nil_of_eq_nil : ∀ {as : List α} {i}, as = [] → as.drop i =
 theorem ne_nil_of_drop_ne_nil {as : List α} {i : Nat} (h: as.drop i ≠ []) : as ≠ [] :=
   mt drop_eq_nil_of_eq_nil h
 
+/-! ### modify head -/
+
+-- Mathlib porting note: List.modifyHead has @[simp], and Lean 4 treats this as
+-- an invitation to unfold modifyHead in any context,
+-- not just use the equational lemmas.
+
+-- @[simp]
+@[simp 1100, nolint simpNF]
```

Comment:
Why is this `nolint` needed? Could you add a comment explaining?

Reply:
Why is the priority high? Does something else trigger first? Perhaps check whether things work removing it, and otherwise add a comment explaining?